### DISCUSSION TITLE
Отображение url в превью

### DISCUSSION THREAD

**Discussion by Qb1cc** [totumonline/totum-ru-issues-and-discussions#163] [Как это сделать?] [answered]
Очень было бы полезно отображать кликабельные ссылки в превью.

![image](https://user-images.githubusercontent.com/84081305/175887985-3c70ba0b-065c-4720-b7a8-a071fabdab3f.png)

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Он так оформляется если превью это поле тип Строка с параметром показывать как url:

<img width="638" alt="Снимок экрана 2022-06-27 в 16 50 19" src="https://user-images.githubusercontent.com/55755565/175957826-be87add7-8b9d-40f7-8f7a-4ed7f61f5833.png">

<img width="593" alt="Снимок экрана 2022-06-27 в 16 50 55" src="https://user-images.githubusercontent.com/55755565/175957864-6d704a34-0450-4a6f-ab7e-3c4777cf7a0f.png">

**Также есть другой путь** — не использовать превью, а использовать [panelHtml](https://ru.docs.totum.online/functions#fn-panelHtml)

**Reply by Qb1cc** [reply]
Спасиб, я почему то прочитав подумал что это только работает для значений полей. PanelHTML крутая вещь, но не совсем подходит мне. Менеджеры проверяя заказ, должны просто быстро без всяких переключений иметь возможность пойти на сайт поставщика, прога работает на горячих продажах и чем быстрее менеджер сможет всё посмотреть , тем быстрее клиент получит свою услугу. Тотум хоть и с боем ( пришлось заставлять переучиваться ) показывает хорошие результаты, но есть конечно моменты такие как печать , на маломощных компах почему то подготовка страницы из браузера достаточно долго идёт.. с чем это связанно не знаю, а так же есть некие глюки в старых браузерах, я всё записываю и потом просто скопом вкину, баги это или нет не знаю. Их не много, но есть.

**Reply by ttm-alexeypolunin** [reply]
Для печати поставлена задержка именно для маломощных компов — иначе будет пустая страница. Со старыми браузерами мы вам помочь не сможем. 

Про `panelHtml` не понял, тк видимый эффект будет тот-же за исключением третьего уровня в выпадающем списке.

<img width="600" alt="Снимок экрана 2022-06-27 в 21 06 12" src="https://user-images.githubusercontent.com/55755565/176007061-bb736361-61a6-464d-9f8e-992828f5050b.png">

**Reply by Qb1cc** [reply]
> Для печати поставлена задержка именно для маломощных компов — иначе будет пустая страница. Со старыми браузерами мы вам помочь не сможем.

а можно узнать минимальные характеристики необходимые, что бы выйти из этой задержки? На стареньких i3 с 6 гигами оперы ddr3  уже заметно, но допустим на r 5 3500x воообще всё работает идеально ( изначально я и тестировал всё на таких компах , и кстате не советую покупать именно 3500 в любом виде, они хоть и вроде не плохие, но сделаны они для России и Китая,  урезали  SMT(одно ядро один вычислительный поток)  и проц стал стоить 150$ вместо 200, потом уже столкнулся с задержкой потому что начали некоторые жаловаться )

**Reply by ttm-alexeypolunin** [reply]
Там как работает — отрисовывается страница, которую вы не видите — она подвинута офсетом за край экрана и потом она отправляется на печать.

Когда мы столкнулись с тем, что иногда не успевает отрисоваться, то поставили задержку 2-4 сек.

Потом выяснилось, что в firefox она получилась 10-12 — это исправили в тройке.

Мне этот `sleep` не очень нравиться — что мы можем сделать — это вынести его настройку в `Conf.php`

**Reply by mihail-cornev** [reply]
> **Также есть другой путь** — не использовать превью, а использовать [panelHtml](https://ru.docs.totum.online/functions#fn-panelHtml)

Алексей, подскажите пожалуйста как правильно делать такие ссылки в панели по правой кнопке?

Например, значение поля это номер телефона; 37367111555

И я хочу по правой кнопке мыши сделать чтобы выходило меню, где можно было бы выбрать ссылку типа viber, telegram, whatsapp. Т.е. чтобы по нажатию на ссылку открывалось нужное приложение и чат с этим номером, но никак не возьму в толк, как правильно работает функция panelhtml. 

![Screensho45_2](https://user-images.githubusercontent.com/113771444/209565725-7324c2fa-56d5-4b3d-a7a2-8744d10515ac.jpg)

UPD: разобрался.

```
p1=: panelHtml(html: str`"<a href='viber://chat?number=" + $phone1 + "' target='_blank'>Viber</a>"`)
p2=: panelHtml(html: str`"<a href='https://wa.me/" + $phone2 + "' target='_blank'>Whatsapp</a>"`)
p3=: panelHtml(html: str`"<a href='https://t.me/" + $phone1 + "' target='_blank'>Telegram</a>"`)
phone1: str`"+" + $#nfv`
phone2: $#nfv
```
